pkg/server/profiles_test.go

```diff
@@ -0,0 +1,182 @@
+				Expect(err).To(BeAssignableToTypeOf(&grpcruntime.HTTPStatusError{}))
+				//TODO why do we return 200 when the HelmRepository doesn't exist
+				Expect(err.(*grpcruntime.HTTPStatusError).HTTPStatus).To(Equal(http.StatusOK))
+			})
+		})
+	})
+
+	Describe("GetProfileValues", func() {
+		When("the HelmRepository exists", func() {
+			BeforeEach(func() {
+				Expect(kubeClient.Create(context.TODO(), helmRepo)).To(Succeed())
+			})
+
+			When("it retrieves the values file from Helm chart", func() {
+				When("header does not contain 'application/octet-stream'", func() {
+					It("returns a values file in base64-encoded json", func() {
+						fakeHelmRepoManager.GetValuesFileReturns([]byte("values"), nil)
+
+						resp, err := s.GetProfileValues(context.TODO(), &pb.GetProfileValuesRequest{
+							ProfileName:    profileName,
+							ProfileVersion: profileVersion,
+						})
+						Expect(err).NotTo(HaveOccurred())
+						Expect(resp.ContentType).To(Equal(server.JsonType))
+						valuesResp := &pb.GetProfileValuesResponse{}
+						err = json.Unmarshal(resp.Data, valuesResp)
+						Expect(err).NotTo(HaveOccurred())
+						profileValues, err := base64.StdEncoding.DecodeString(valuesResp.Values)
+						Expect(err).NotTo(HaveOccurred())
+						Expect(string(profileValues)).To(Equal("values"))
+					})
+				})
+
+				When("header contains 'application/octet-stream'", func() {
+					It("returns a values file in binary", func() {
+						fakeHelmRepoManager.GetValuesFileReturns([]byte("values"), nil)
+						ctx := metadata.NewIncomingContext(context.Background(), metadata.Pairs("accept", server.OctetStreamType))
+
+						resp, err := s.GetProfileValues(ctx, &pb.GetProfileValuesRequest{
+							ProfileName:    profileName,
+							ProfileVersion: profileVersion,
+						})
+						Expect(err).NotTo(HaveOccurred())
+						Expect(resp.ContentType).To(Equal(server.OctetStreamType))
+						Expect(string(resp.Data)).To(Equal("values"))
+					})
+				})
+
+				When("it cannot retrieve the values file from Helm chart", func() {
+					It("errors", func() {
+						fakeHelmRepoManager.GetValuesFileReturns([]byte{}, fmt.Errorf("err"))
+						_, err := s.GetProfileValues(context.TODO(), &pb.GetProfileValuesRequest{
+							ProfileName:    profileName,
+							ProfileVersion: profileVersion,
+						})
+						Expect(err).To(MatchError("failed to retrieve values file from Helm chart 'observability' (latest): err"))
+					})
+				})
+			})
+		})
+
+		When("the HelmRepository doesn't exist", func() {
+			It("errors", func() {
+				_, err := s.GetProfileValues(context.TODO(), &pb.GetProfileValuesRequest{})
+				Expect(err).To(MatchError("HelmRepository \"default\"/\"helmrepo\" does not exist"))
+				Expect(err).To(BeAssignableToTypeOf(&grpcruntime.HTTPStatusError{}))
+				//TODO why do we return 200 when the HelmRepository doesn't exist
+				Expect(err.(*grpcruntime.HTTPStatusError).HTTPStatus).To(Equal(http.StatusOK))
```

Comment:
We have this thing that might be of use: https://github.com/weaveworks/weave-gitops/blob/ee12fdf5957a9070994295e70ec5c2320cc3106f/pkg/server/grpc_error_matcher.go#L25

Reply:
that seems to be testing the grpc status codes, as opposed to http status codes. I hadn't seen the grpc codes before which is interesting :thinking: